### PR TITLE
Add initconfig support for command defaults, allow strings as config keys

### LINKED ISSUE #63: does not store default_value for command flags
using version `1.4.0`

if I run `progname initconfig` on this code, the default value for `other` will not be stored for `command`

``` ruby
#!/usr/bin/env ruby
require 'gli'
include GLI

config_file '/tmp/cfg.yml'

desc 'specify a value to use.'
default_value 'value shows in cfg file twice'
flag [:v,:value]
arg_name 'value'

desc 'command to show default values dont store for command flags'
command :command do |c|
  c.desc 'command value'
  c.default_value 'will not store in cfg'
  c.flag [:o,:other]
end
exit run(ARGV)
```

show help output that default value is set correctly

```
$ ./test.rb help command
command [command options] value
    command to show default values dont store for command flags

Command Options:
    -o, --other=arg - command value (default: will not store in cfg)
```

show it not being stored in cfg file

```
 cat /tmp/cfg.yml 
--- 
:v: value shows in cfg file twice
:value: value shows in cfg file twice
commands: 
  :command: {}
```

### PR DESCRIPTION
**Pull request by hyperbolist**
Fixes #63

The `initconfig` command will now write out all of the switches and flags for any commands and any subcommands.  It writes config keys as strings, though symbols will still work.

### CODE REVIEW

File: lib/gli/monkeypatches/with_indifferent_access.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# adapted from https://gist.github.com/Integralist/9503099
+class Object
```
Comment (davetron5000) [inline lib/gli/monkeypatches/with_indifferent_access.rb]: Can we not monkeypatch core classes?  This is likely going to break anyone that uses GLI with Rails/ActiveSupport.

Comment (davetron5000) [inline lib/gli/monkeypatches/with_indifferent_access.rb]: Looking over the rest of the PR, I think a better way to do this is make a class whose job is to load the configuration and have it do all this string/symbol copying, so we don't infect every object in the entire system with a method that's only needed in a couple of places.

**Comment by davetron5000** [comment]
Thanks for doing this!  Other than the issues with monkeypatching, the overall change looks great!

File: lib/gli/monkeypatches/with_indifferent_access.rb (hunk 1 above)
Comment (hyperbolist) [inline lib/gli/monkeypatches/with_indifferent_access.rb]: Yes, that's fair.  It hadn't occurred to me that a GLI project might also have ActiveSupport loaded.

**Comment by hyperbolist** [comment]
Got rid of the monkeypatching, which I admit was reckless.

File: lib/gli/config_loader.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+module GLI
+  class ConfigLoader
+    def self.load(path)
+      File.open(path) { |file| apply_indifferent_access_to(YAML::load(file)) }
+    end
+
+    # adapted from https://gist.github.com/Integralist/9503099
+    def self.apply_indifferent_access_to(obj)
+      return obj.reduce({}) do |memo, (k, v)|
```
Comment (davetron5000) [inline lib/gli/config_loader.rb]: Sorry to nitpick, but can you use a more traditional `if` construct?  Trailing `if`s for multiple lines is really confusing to me.

**Comment by davetron5000** [comment]
Thanks and sorry for the delay in looking at this.  Just one tiny nitpick, but otherwise this looks good.  Thanks!

**Comment by davetron5000** [comment]
Closing due to being very old. Sorry - looks like I dropped the ball

**Pull request closed without merging**